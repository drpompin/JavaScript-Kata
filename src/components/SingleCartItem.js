import Image from 'next/image'
import React, { useContext } from 'react'
import { KataContext } from '../../context'
import { CurrencyFormatter } from '../utilities/currencyFormatter'
import ButtonComponent from './Button'


export default function SingleCartItem({ kataItem }) {
    const { handleAddToCart, handleDecreaseItem, handleRemoveItem } = useContext(KataContext);
    
    return (
        <div key={kataItem?.name} className="rounded-lg border p-5" data-testid="singleCartItem">
            <div className="flex items-center gap-4 sm:gap-8">
                <Image src={kataItem?.image} alt='item-image' className='w-[30%]' />

                <div>
                    <div className='text-[#27abce] mt-2 font-semibold'>{kataItem?.name}</div>
                    <div className='font-semibold text-sm my-2'>Quantity: {kataItem?.qty}</div>
                    <div className='font-semibold text-sm my-2'>Unit Price: {CurrencyFormatter(kataItem?.unitPrice)}</div>
                    <div className='font-semibold text-sm my-2'>Total Cost: {CurrencyFormatter(kataItem?.total)}</div>
                </div>
            </div>

            <div className="mt-4 flex gap-2">
                <ButtonComponent 
                    title="Inc +"
                    onClick={() => handleAddToCart(kataItem)}
                    style="px-4 py-1 w-fit" 
                    dataTestid="increaseCartItem"
                />
                <ButtonComponent 
                    title="Dec -"
                    onClick={() => handleDecreaseItem(kataItem)}
                    style="px-4 py-1 w-fit" 
                    disabled={kataItem?.qty === 1}
                    dataTestid="decreaseCartItem"
                />
                <ButtonComponent 
                    title="Remove x"
                    onClick={() => handleRemoveItem(kataItem)}
                    style="px-4 py-1 w-fit" 
                    dataTestid="removeCartItem"
                />
            </div>
        </div>
    )
}
