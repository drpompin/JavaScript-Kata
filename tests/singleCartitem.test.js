import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import SingleCartItem from "../src/components/SingleCartItem";

const kataCartItemMock = {
    image: {src: '/_next/static/media/hawaiian-pizza.051da743.jpg'},
    name: "Item A",
    qty: 1,
    specialPrice: "3 for 130",
    total: 50,
    unitPrice: 50
}

describe('Single Cart Item Component', () => {
test('renders single cart item correctly', () => {
    const { asFragment } = render(
      <SingleCartItem item={kataCartItemMock} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
})
