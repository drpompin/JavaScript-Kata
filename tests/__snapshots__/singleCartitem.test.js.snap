// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Cart Item Component renders single cart item correctly 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border p-5"
    data-testid="singleCartItem"
  >
    <div
      class="flex items-center gap-4 sm:gap-8"
    >
      <img
        alt="item-image"
        class="w-[30%]"
        data-nimg="1"
        decoding="async"
        loading="lazy"
        src=""
        style="color: transparent;"
      />
      <div>
        <div
          class="text-[#27abce] mt-2 font-semibold"
        />
        <div
          class="font-semibold text-sm my-2"
        >
          Quantity: 
        </div>
        <div
          class="font-semibold text-sm my-2"
        >
          Unit Price: £NaN
        </div>
        <div
          class="font-semibold text-sm my-2"
        >
          Total Cost: £NaN
        </div>
      </div>
    </div>
    <div
      class="mt-4 flex gap-2"
    >
      <button
        class="w-full py-2 font-semibold border-none rounded-md text-sm bg-[#f52421] hover:bg-[#ef3b39] text-white px-4 py-1 w-fit "
        data-testid="increaseCartItem"
      >
        Inc +
      </button>
      <button
        class="w-full py-2 font-semibold border-none rounded-md text-sm bg-[#f52421] hover:bg-[#ef3b39] text-white px-4 py-1 w-fit "
        data-testid="decreaseCartItem"
      >
        Dec -
      </button>
      <button
        class="w-full py-2 font-semibold border-none rounded-md text-sm bg-[#f52421] hover:bg-[#ef3b39] text-white px-4 py-1 w-fit "
        data-testid="removeCartItem"
      >
        Remove x
      </button>
    </div>
  </div>
</DocumentFragment>
`;
